### PR TITLE
fix: the test database is not cleaned up from the Cosmos account when integration tests fail

### PR DESCRIPTION
**Pull request by Angith**
The test database was not getting deleted while following cases
1. if the test fails by exceeding the total throughput of the Cosmos account. 
2. if the test panics.
3. If the user interrupts the test. 

The PR tries to tear down all the test data created for the above cases. 

**Exception**: Cleaning up is not feasible for the panic errors that happen during `m.Run()` in `TestMain`. Check this issue for more details: https://github.com/golang/go/issues/37206

### CODE REVIEW

File: instrumentation/instacosmos/container_client_test.go [hunk 1]
```diff
@@ -65,6 +67,9 @@ var (
 	rec               *instana.Recorder
 )
 
+// exit error code
+var exitNotOk int = 1
```
Comment (nithinputhenveettil) [inline instrumentation/instacosmos/container_client_test.go:71]: Better to define as a `const`?

File: instrumentation/instacosmos/container_client_test.go [hunk 2]
```diff
@@ -168,6 +194,18 @@ func TestMain(m *testing.M) {
 	tracer := instana.NewTracerWithEverything(&instana.Options{AgentClient: alwaysReadyClient{}}, recorder)
 	sensor := instana.NewSensorWithTracer(tracer)
 
+	// handles panic errors
+	// this will only work for the panic errors before m.Run()
+	// check this issue for more details: https://github.com/golang/go/issues/37206
+	defer func() {
+		if err := recover(); err != nil {
+			shutdown(sensor, exitNotOk)
```
Comment (nithinputhenveettil) [inline instrumentation/instacosmos/container_client_test.go:202]: Isn't it better to re-throw the panic after calling shutdown ?

**Review by nithinputhenveettil — CHANGES_REQUESTED** [review]

File: instrumentation/instacosmos/container_client_test.go (hunk 1 above)
Comment (Angith) [reply instrumentation/instacosmos/container_client_test.go]: done 👍

File: instrumentation/instacosmos/container_client_test.go (hunk 2 above)
Comment (Angith) [reply instrumentation/instacosmos/container_client_test.go]: yes. Good Catch 👍

**Review by nithinputhenveettil — APPROVED** [review]
:shipit:

**Review by sanojsubran — APPROVED** [review]

**Pull request merged**